functions/Set-DbaJobOutputFile.ps1

```diff
@@ -0,0 +1,139 @@
+﻿
+Function Set-DbaJobOutPutFile
+{
+<#
+.Synopsis
+   Sets the OutPut File for a step of an agent job with the Job Names and steps provided dynamically 
+.DESCRIPTION
+   Sets the OutPut File for a step of an agent job with the Job Names and steps provided dynamically if required
+
+.PARAMETER SqlServer
+    The SQL Server that you're connecting to.
+
+.PARAMETER SQLCredential
+    Credential object used to connect to the SQL Server as a different user be it Windows or SQL Server. Windows users are determiend by the existence of a backslash, so if you are intending to use an alternative Windows connection instead of a SQL login, ensure it contains a backslash.
+
+.PARAMETER JobName
+    The Agent Job Name to provide Output File Path for. Also available dynamically
+
+.PARAMETER JobStep
+    The Agent Job Step to provide Output File Path for. Also available dynamically
+
+.PARAMETER OutputFile
+    The Full Path to the New Output file
+.EXAMPLE
+   Set-DbaJobOutPutFile -sqlserver SERVERNAME -JobName 'The Agent Job' -OutPutFile E:\Logs\AgentJobStepOutput.txt
+
+   Sets the Job step for The Agent job on SERVERNAME to E:\Logs\AgentJobStepOutput.txt
+.NOTES
+   AUTHOR - Rob Sewell https://sqldbawithabeard.com
+   DATE - 30/10/2016
+#>
+[CmdletBinding(SupportsShouldProcess = $true)]
+param
+(
+        [Parameter(Mandatory=$true,HelpMessage='The SQL Server Instance', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=0)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [string]$SqlServer,
+        [Parameter(Mandatory=$false,HelpMessage='SQL Credential', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=1)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory=$true,HelpMessage='The Full Output File Path', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=2)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [string]$OutputFile,
+        [Parameter(Mandatory=$false,HelpMessage='The Job Step name', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [object]$JobStep)
+
+    DynamicParam {
```

Comment:
There is already a built-in dynamic param for this (that will soon be migrated to TabExpansion++. Please see Copy-SqlJob or probably Test-DbaJobOwner for the syntax.